dali/pipeline/data/types_test.cc

```diff
@@ -84,4 +84,35 @@ TYPED_TEST(TypesTest, TestRegisteredType) {
   ASSERT_EQ(type.name(), this->TypeName());
 }
 
+#define GET_TYPE_INFO(Type) \
+const auto &ANONYMIZE_VARIABLE(info) = TypeTable::GetTypeInfo(type2id<Type>::value)
+
+TEST(TypeTableTest, BasicTypesLookup) {
```

Comment:
This test doesn't really test anything, since it's not guaranteed that it's invoked before other tests that might register types as a side effect. To that effect, you should put the results of GetTypeInfo in _global_ variables and check them in the test case, e.g.:
```
template <typename T>
auto GetTypeInfo() {
  return make_pair(type2id<T>::value, &TypeTable::GetTypeInfo(type2id<T>::value));
}

static std::pair<DALIDataType, const TypeInfo *> g_types[] = {
GetTypeInfo<uint8_t>(),
GetTypeInfo<int8_t>(),
GetTypeInfo<uint32_t>(),
GetTypeInfo<bool>(),
GetTypeInfo<float>(),
GetTypeInfo<double>(),
};

TEST(TypeTableTest, BasicTypesLookup) {
  for (auto &info : g_types) {
    ASSERT_NE(info.second, nullptr);
    EXPECT_EQ(info.first, info.second->id());
  }
}
```